syft/pkg/cataloger/python/parse_uv_lock.go

```diff
@@ -0,0 +1,242 @@
+package python
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/internal/unknown"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+	"github.com/anchore/syft/syft/pkg/cataloger/internal/dependency"
+)
+
+// We use this to check for the version before we try to parse.
+// The TOML library handily ignores everything that isn't mentioend in the struct.
+type UvLockFileVersion struct {
```

Comment:
any structs that are introduced in this package should probably be unexported (start with a lowercase)